flocker/control/test/test_persistence.py

```diff
@@ -249,3 +251,24 @@ def test_complex_keys(self):
                                manifestations={}, paths={},
                                devices={uuid4(): FilePath(b"/tmp")})
         self.assertEqual(node_state, wire_decode(wire_encode(node_state)))
+
+
+class ConfigurationMigrationTests(SynchronousTestCase):
+    """
+    Tests for ``_ConfigurationMigration``.
+    """
+    configurations_dir = FilePath(__file__).sibling('configurations')
+
+    def test_v1_latest_configuration(self):
+        """
+        A V1 JSON configuration blob is transformed to the latest
+        configuration, with the result validating when loaded.
+        """
+        v1_json = self.configurations_dir.child(
+            b"configuration_v1.json").getContent()
+        upgraded_json = migrate_configuration(1, 2, v1_json)
+        upgraded_config = wire_decode(upgraded_json)
+        expected_configuration = Configuration(
+            version=_CONFIG_VERSION, deployment=TEST_DEPLOYMENT
```

Comment:
Possibly. although an assertion on TEST_DEPLOYMENT or an equivalent is something we *want* to break if we change the Deployment model, or any of the other models it encapsulates.